### PR TITLE
Add support for the crkbd r2g by mechboards

### PR DESCRIPTION
**Pull request by e11i0t23**
## Description

Add support for the crkbd r2g by Mechboards UK, a variation on the corne which is factory assembled, requries a few adjustments to current code, so implemented the crkbd/r2g, which enables compatibility between keymaps ect

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/crkbd/r2g/r2g.c [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+    if ((keycode >= QK_MOD_TAP && keycode <= QK_MOD_TAP_MAX) ||
+        (keycode >= QK_LAYER_TAP && keycode <= QK_LAYER_TAP_MAX)) { keycode = keycode & 0xFF; }
+  if (keycode < 60) {
+    name = code_to_name_r2g[keycode];
+  }
+
+  // update keylog
+  snprintf(keylog_str_r2g, sizeof(keylog_str_r2g), "%dx%d, k%2d : %c",
+           record->event.key.row, record->event.key.col,
+           keycode, name);
```
Comment (drashna) [inline keyboards/crkbd/r2g/r2g.c:143]: This should be converted to NOT use snprintf, as it takes 1.5k, and `get_u8_str` can generate a string without using snprintf.

File: keyboards/crkbd/r2g/r2g.c [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+};
+    oled_write_raw_P(mb_logo, sizeof(mb_logo));
+    //oled_set_cursor(oled_max_chars()/2,oled_max_lines()/2);
+    //oled_write_P(PSTR("R2G"), false);
+}
+
+__attribute__((weak)) void oled_task_user(void) {
```
Comment (drashna) [inline keyboards/crkbd/r2g/r2g.c:205]:
```suggestion
bool oled_task_kb(void) {
    if (!oled_task_user()) { return false; }
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by e11i0t23** [comment]
Resolved to update in line with develop merge and suggestions

File: keyboards/crkbd/r2g/config.h [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+#   define RGBLIGHT_EFFECT_RGB_TEST
+#   define RGBLIGHT_EFFECT_ALTERNATING
+#   define RGBLIGHT_EFFECT_TWINKLE
+
+#   define RGBLED_NUM 54
+#   define RGBLED_SPLIT \
+        { 27, 27 }
+#endif
+
+#ifdef RGB_MATRIX_ENABLE
```
Comment (drashna) [inline keyboards/crkbd/r2g/config.h:46]: Just a heads up, RGB Matrix now requires explicit enabling of animation modes.  Any mode not enabled by a define will not be enabled. 

```c
#define ENABLE_RGB_MATRIX_ALPHAS_MODS
#define ENABLE_RGB_MATRIX_GRADIENT_UP_DOWN
#define ENABLE_RGB_MATRIX_GRADIENT_LEFT_RIGHT
#define ENABLE_RGB_MATRIX_BREATHING
#define ENABLE_RGB_MATRIX_BAND_SAT
#define ENABLE_RGB_MATRIX_BAND_VAL
#define ENABLE_RGB_MATRIX_BAND_PINWHEEL_SAT
#define ENABLE_RGB_MATRIX_BAND_PINWHEEL_VAL
#define ENABLE_RGB_MATRIX_BAND_SPIRAL_SAT
#define ENABLE_RGB_MATRIX_BAND_SPIRAL_VAL
#define ENABLE_RGB_MATRIX_CYCLE_ALL
#define ENABLE_RGB_MATRIX_CYCLE_LEFT_RIGHT
#define ENABLE_RGB_MATRIX_CYCLE_UP_DOWN
#define ENABLE_RGB_MATRIX_RAINBOW_MOVING_CHEVRON
#define ENABLE_RGB_MATRIX_CYCLE_OUT_IN
#define ENABLE_RGB_MATRIX_CYCLE_OUT_IN_DUAL
#define ENABLE_RGB_MATRIX_CYCLE_PINWHEEL
#define ENABLE_RGB_MATRIX_CYCLE_SPIRAL
#define ENABLE_RGB_MATRIX_DUAL_BEACON
#define ENABLE_RGB_MATRIX_RAINBOW_BEACON
#define ENABLE_RGB_MATRIX_RAINBOW_PINWHEELS
#define ENABLE_RGB_MATRIX_RAINDROPS
#define ENABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS
#define ENABLE_RGB_MATRIX_HUE_BREATHING
#define ENABLE_RGB_MATRIX_HUE_PENDULUM
#define ENABLE_RGB_MATRIX_HUE_WAVE
#define ENABLE_RGB_MATRIX_PIXEL_FRACTAL
#define ENABLE_RGB_MATRIX_PIXEL_RAIN

#define ENABLE_RGB_MATRIX_TYPING_HEATMAP
#define ENABLE_RGB_MATRIX_DIGITAL_RAIN

#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_SIMPLE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_WIDE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTIWIDE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
#define ENABLE_RGB_MATRIX_SPLASH
#define ENABLE_RGB_MATRIX_MULTISPLASH
#define ENABLE_RGB_MATRIX_SOLID_SPLASH
#define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
```

File: keyboards/crkbd/r2g/r2g.c [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+    2, 2, 2, 2, 2, 1, 4,
+    4, 4, 4, 4, 4, 1, 1,
+    4, 4, 4, 4, 4, 4, 4,
+    4, 4, 1, 1, 1
+} };
+
+#endif
+
+#ifdef OLED_ENABLE
+#include <stdio.h>
```
[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/crkbd/r2g/r2g.c:91]: This should only be needed if snprintf is being used.  If that's been removed, then this is no longer needed.

**Review by drashna — CHANGES_REQUESTED** [review]
please remove `lib/ugfx`

File: keyboards/crkbd/r2g/r2g.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (e11i0t23) [reply keyboards/crkbd/r2g/r2g.c]: removed in 74b0603

**Comment by e11i0t23** [comment]
Defined a range of matrix modes that fit whilst leaving plenty of space when enabled

**Review by drashna — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Pull request merged**